### PR TITLE
tree-wide: use CLEANUP_ERASE() at many places

### PR DESCRIPTION
**Pull request by poettering**
We now have this nice construct, let's use it.

Follow-up for: #25973

### CODE REVIEW

File: src/fundamental/memory-util-fundamental.h [hunk 1]
```diff
@@ -38,5 +38,12 @@ static inline void erase_var(struct VarEraser *e) {
 }
 
 /* Mark var to be erased when leaving scope. */
-#define CLEANUP_ERASE(var) \
-        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { .p = &var, .size = sizeof(var) }
+#define CLEANUP_ERASE(var)                                              \
+        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { \
+                .p = &var, .size = sizeof(var)                          \
+        }
```
Comment (medhefgo) [inline src/fundamental/memory-util-fundamental.h]:
```suggestion
#define CLEANUP_ERASE(var) CLEANUP_ERASE_PTR(&var, sizeof(var))
```

Comment (poettering) [reply src/fundamental/memory-util-fundamental.h]: right, of course!

**Comment by poettering** [comment]
force pushed a new version. ptal!

**Review by medhefgo — APPROVED** [review]
Btw, why was c chosen instead of cpp to start systemd? It would've saved everyone so many headaches…

**Comment by poettering** [comment]
none of the people originally involved (me included) think C++ is a good language.

(We have discussed Rust over and over again though, and I think everyone of the usual peeps is onboard with going to Rust sooner or later. The Meson vs. Cargo mess is kinda stopping us from fully committing, since we'll have a hybrid codebase for a long long time before we can leave C behind.)

**Comment by bluca** [comment]
CI is sad

```
../src/shared/tpm2-util.c
../src/shared/tpm2-util.c:1442:17: error: cannot jump from this goto statement to its label
                goto finish;
                ^
../src/shared/tpm2-util.c:1464:9: note: jump bypasses initialization of variable with __attribute__((cleanup))
        CLEANUP_ERASE(hmac_sensitive);
        ^
../src/fundamental/memory-util-fundamental.h:47:28: note: expanded from macro 'CLEANUP_ERASE'
#define CLEANUP_ERASE(var) CLEANUP_ERASE_PTR(&(var), sizeof(var))
                           ^
../src/fundamental/memory-util-fundamental.h:41:56: note: expanded from macro 'CLEANUP_ERASE_PTR'
        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { \
                                                       ^
../src/fundamental/macro-fundamental.h:68:27: note: expanded from macro 'CONCATENATE'
#define CONCATENATE(x, y) XCONCATENATE(x, y)
                          ^
../src/fundamental/macro-fundamental.h:67:28: note: expanded from macro 'XCONCATENATE'
#define XCONCATENATE(x, y) x ## y
                           ^
<scratch space>:79:1: note: expanded from here
_eraser_15
^
../src/shared/tpm2-util.c:1425:17: error: cannot jump from this goto statement to its label
                goto finish;
                ^
../src/shared/tpm2-util.c:1464:9: note: jump bypasses initialization of variable with __attribute__((cleanup))
        CLEANUP_ERASE(hmac_sensitive);
        ^
../src/fundamental/memory-util-fundamental.h:47:28: note: expanded from macro 'CLEANUP_ERASE'
#define CLEANUP_ERASE(var) CLEANUP_ERASE_PTR(&(var), sizeof(var))
                           ^
../src/fundamental/memory-util-fundamental.h:41:56: note: expanded from macro 'CLEANUP_ERASE_PTR'
        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { \
                                                       ^
../src/fundamental/macro-fundamental.h:68:27: note: expanded from macro 'CONCATENATE'
#define CONCATENATE(x, y) XCONCATENATE(x, y)
                          ^
../src/fundamental/macro-fundamental.h:67:28: note: expanded from macro 'XCONCATENATE'
#define XCONCATENATE(x, y) x ## y
                           ^
<scratch space>:79:1: note: expanded from here
_eraser_15

```

**Review by keszybz — COMMENTED** [review]
LGTM.

**Comment by keszybz** [comment]
```
../src/shared/tpm2-util.c:1441:17: error: cannot jump from this goto statement to its label
                goto finish;
                ^
../src/shared/tpm2-util.c:1463:9: note: jump bypasses initialization of variable with __attribute__((cleanup))
        CLEANUP_ERASE(hmac_sensitive);
        ^
../src/fundamental/memory-util-fundamental.h:47:28: note: expanded from macro 'CLEANUP_ERASE'
#define CLEANUP_ERASE(var) CLEANUP_ERASE_PTR(&(var), sizeof(var))
                           ^
../src/fundamental/memory-util-fundamental.h:41:56: note: expanded from macro 'CLEANUP_ERASE_PTR'
        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { \
                                                       ^
../src/fundamental/macro-fundamental.h:68:27: note: expanded from macro 'CONCATENATE'
#define CONCATENATE(x, y) XCONCATENATE(x, y)
                          ^
../src/fundamental/macro-fundamental.h:67:28: note: expanded from macro 'XCONCATENATE'
#define XCONCATENATE(x, y) x ## y
                           ^
<scratch space>:79:1: note: expanded from here
_eraser_15
^
../src/shared/tpm2-util.c:1424:17: error: cannot jump from this goto statement to its label
                goto finish;
                ^
../src/shared/tpm2-util.c:1463:9: note: jump bypasses initialization of variable with __attribute__((cleanup))
        CLEANUP_ERASE(hmac_sensitive);
        ^
../src/fundamental/memory-util-fundamental.h:47:28: note: expanded from macro 'CLEANUP_ERASE'
#define CLEANUP_ERASE(var) CLEANUP_ERASE_PTR(&(var), sizeof(var))
                           ^
../src/fundamental/memory-util-fundamental.h:41:56: note: expanded from macro 'CLEANUP_ERASE_PTR'
        _cleanup_(erase_var) _unused_ struct VarEraser CONCATENATE(_eraser_, UNIQ) = { \
                                                       ^
../src/fundamental/macro-fundamental.h:68:27: note: expanded from macro 'CONCATENATE'
#define CONCATENATE(x, y) XCONCATENATE(x, y)
                          ^
../src/fundamental/macro-fundamental.h:67:28: note: expanded from macro 'XCONCATENATE'
#define XCONCATENATE(x, y) x ## y
                           ^
<scratch space>:79:1: note: expanded from here
_eraser_15
^
2 errors generated.
```

File: src/shared/tpm2-util.c [hunk 2]
```diff
@@ -1455,10 +1455,13 @@ int tpm2_seal(const char *device,
                 },
         };
 
-        hmac_sensitive = (TPM2B_SENSITIVE_CREATE) {
+        TPM2B_SENSITIVE_CREATE hmac_sensitive = (TPM2B_SENSITIVE_CREATE) {
                 .size = sizeof(hmac_sensitive.sensitive),
                 .sensitive.data.size = 32,
         };
+
+        CLEANUP_ERASE(hmac_sensitive);
```
Comment (keszybz) [inline src/shared/tpm2-util.c]: I guess the `goto` must not jump over this initializer…

**Comment by bluca** [comment]
can you push again? the tests somehow got lost

**Comment by bluca** [comment]
> can you push again? the tests somehow got lost

done it for you

**Comment by keszybz** [comment]
CI has two kinds of failures:

00:52:18 systemd-networkd_sanitizers
00:52:18 TEST-54-CREDS

00:48:56 systemd-networkd-tests.py
00:48:56 TEST-54-CREDS

01:38:03 systemd-networkd-tests.py
01:38:03 TEST-54-CREDS

TEST-54-CREDS:                      FAIL     (  3 s)

subprocess.CalledProcessError: Command '['/systemd-meson-build/systemd-networkd-wait-online', '--timeout=20s', '--interface=wg99:routable', '--interface=wg98:routable', '--interface=wg97:carrier', '--ignore=ip_vti0', '--ignore=tun99', '--ignore=ifb1', '--ignore=ifb0', '--ignore=gretap0', '--ignore=tap99', '--ignore=ip6tnl0', '--ignore=ip6gre0', '--ignore=teql0', '--ignore=eth0', '--ignore=lo', '--ignore=tunl0', '--ignore=ip6_vti0', '--ignore=gre0', '--ignore=sit0', '--ignore=erspan0']' returned non-zero exit status 1.

[ 3095.740552] systemd[1]: Assertion failed for /usr/bin/true.
[ 3095.774435] systemd[1]: run-u8.service: Failed with result 'exit-code'.
[ 3095.827814] systemd[1]: run-u9.service: Failed with result 'exit-code'.
[ 3095.904823] systemd-creds[91]: Credential secret file '/var/lib/systemd/credential.secret' is not located on encrypted media, using anyway.
[ 3095.950336] systemd[97]: Unknown encryption format, or corrupted data: Operation not supported
[ 3095.950396] systemd[96]: run-u11.service: Failed to set up credentials: Protocol error
[ 3095.950433] systemd[96]: run-u11.service: Failed at step CREDENTIALS spawning /usr/bin/cat: Protocol error
[ 3095.951704] systemd[1]: run-u11.service: Failed with result 'exit-code'.
[ 3095.952163] systemd[1]: testsuite-54.service: Failed with result 'exit-code'.
[ 3095.952300] systemd[1]: Failed to start TESTSUITE-54-CREDS.
[ 3096.070934] systemd[1]: run-credentials-run\x2du0.service.mount: Failed with result 'exit-code'.
[ 3096.074041] systemd[1]: run-credentials-systemd\x2dtmpfiles\x2dsetup.service.mount: Failed with result 'exit-code'.

Looks suspicious.

**Comment by keszybz** [comment]
Those failures don't appear in other PRs. I'll set `ci-fails`.

**Comment by poettering** [comment]
pushed a new version, hopefully addressed the CI issue. ptal.

**Review by yuwata — APPROVED** [review]

**Pull request merged**